wecs/mechanics/clock.py

```diff
@@ -33,14 +33,17 @@ class Clock:
 
 @Component()
 class Calendar:
-    year:   int = 3000
-    #                             list(current, max)
-    month:  list = field(default_factory=lambda: list((1,12)))
-    day:    list = field(default_factory=lambda: list((1,30)))
-    hour:   list = field(default_factory=lambda: list((12,24)))
-    minute: list = field(default_factory=lambda: list((0,60)))
-    second: list = field(default_factory=lambda: list((0,60)))
-    tps:    list = field(default_factory=lambda: list((0,0.003))) #(game) time per second
+    timeframes: dict = field(default_factory=lambda: {
+        "second": [0,59], # in a minute
+        "minute": [0,59], # in an hour
+        "hour":   [12,23],# in a day
+        "day":    [1,30], # in a month
+        "month":  [1,11], # in a year
+        "year":   [2084], # last timeframe doesn't need a maximum
+    })
```

Comment:
Using a dict here means that the code will have to be aware of the existing time units, their names, and their order. Using a list instead would make the names merely informative (and redundant), and would inherently define an ordering.